File: server/LightDeploy.Service/Dto/DeployTargetDto.cs
using LightApi.Infra.ModelValidator;

namespace LightDeploy.Service.Dto;

public class DeployTargetDto
{
    public int Id { get; set; }
    
    /// <summary>
    /// Agent Ip
    /// </summary>
    public string Host { get; set; }
    
    /// <summary>
    /// 服务端口
    /// </summary>
    public int ServicePort { get; set; }
    
    /// <summary>
    /// Agent端口号
    /// </summary>
    public string Port { get; set; }
    
    /// <summary>
    /// 健康检查Url
    /// </summary>
    public string? HealthCheckUrl { get; set; }
    
    /// <summary>
    /// Token
    /// </summary>
    public string? AuthKey { get; set; }
    
    /// <summary>
    /// 服务Id
    /// </summary>
    [RequiredEx]
    public int? ServiceId { get; set; }
}

File: web/src/dtos/deployTargetDto.ts
export interface DeployTargetDto {
  id?: number|undefined;

  /**
   * Agent Ip
   */
  host?: string|undefined;

  /**
   * Agent端口号
   */
  port?: string|undefined;

  /**
   * 健康检查Url
   */
  healthCheckUrl?: string|undefined; // 使用问号表示可选属性

  /**
   * Token
   */
  authKey?: string|undefined; // 使用问号表示可选属性

  /**
   * 服务Id
   */
  serviceId?: number | null |undefined; // 使用 number | null 表示可以为 null 的数字类型

  status?:string|null|undefined;// 状态
}
